TST: linalg.{lu_}solve: `TestBatch` tolerance failures with Intel LLVM compilers 2024.2

### Describe your issue.

I installed scipy and numpy from source using Intel LLVM compilers 2024.2. My `scipy.test()` has 2 failing tests.

### Reproducing Code Example

```python
python3 -c 'import scipy; scipy.test()'
```

### Error message

```shell
=================================== FAILURES ===================================
____________________ TestBatch.test_solve[complex64-bdim2] _____________________
/discover/nobackup/projects/tm5var4d/packages/python/3.13.2/lib/python3.13/site-packages/scipy/linalg/tests/test_batch.py:445: in test_solve
    assert_allclose(x, np.linalg.solve(A, b), atol=2e-6)
E   AssertionError:
E   Not equal to tolerance rtol=1e-07, atol=2e-06
E
E   Mismatched elements: 1 / 120 (0.833%)
E   Max absolute difference among violations: 2.5788913e-06
E   Max relative difference among violations: 4.9293413e-07
E    ACTUAL: array([[[[-0.125288+1.778756e-03j,  0.502625+8.413855e-01j,
E              0.621248-4.769069e-01j,  0.829906-3.271966e-01j],
E            [ 0.403593+3.780714e-01j,  0.280962+3.392008e-02j,...
E    DESIRED: array([[[[-0.125288+1.778829e-03j,  0.502624+8.413856e-01j,
E              0.621248-4.769069e-01j,  0.829906-3.271965e-01j],
E            [ 0.403593+3.780714e-01j,  0.280962+3.392000e-02j,...
        A          = array([[[[0.8561369 +0.05969837j, 0.37626302+0.36519146j,
          0.6539293 +0.40403795j, 0.16993557+0.79370755j,
  ...1648j,
          0.50310206+0.6898472j , 0.6573576 +0.9962738j ,
          0.2530085 +0.86227924j]]]], dtype=complex64)
        b          = array([[[[0.04875669+0.31144133j, 0.45005864+0.22624582j,
          0.797893  +0.5535813j , 0.41498896+0.6063427j ],
 ...94375426j, 0.05030875+0.60288817j,
          0.673049  +0.02498303j, 0.7586571 +0.54190135j]]]],
      dtype=complex64)
        bdim       = (2, 3, 5, 4)
        dtype      = <class 'numpy.complex64'>
        rng        = Generator(PCG64) at 0x14F7FE859E00
        self       = <scipy.linalg.tests.test_batch.TestBatch object at 0x14f888f54230>
        x          = array([[[[-0.12528789+1.7787557e-03j,  0.5026245 +8.4138554e-01j,
           0.6212484 -4.7690693e-01j,  0.829906  -3....0.14644761+6.8131626e-01j,
          -0.91509163+8.1870073e-01j,  0.05595047-2.2421232e-01j]]]],
      dtype=complex64)
___________________ TestBatch.test_lu_solve[complex64-bdim2] ___________________
/discover/nobackup/projects/tm5var4d/packages/python/3.13.2/lib/python3.13/site-packages/scipy/linalg/tests/test_batch.py:459: in test_lu_solve
    assert_allclose(x, np.linalg.solve(A, b), atol=2e-6)
E   AssertionError:
E   Not equal to tolerance rtol=1e-07, atol=2e-06
E
E   Mismatched elements: 1 / 120 (0.833%)
E   Max absolute difference among violations: 2.5788913e-06
E   Max relative difference among violations: 4.9293413e-07
E    ACTUAL: array([[[[-0.125288+1.778756e-03j,  0.502625+8.413855e-01j,
E              0.621248-4.769069e-01j,  0.829906-3.271966e-01j],
E            [ 0.403593+3.780714e-01j,  0.280962+3.392008e-02j,...
E    DESIRED: array([[[[-0.125288+1.778829e-03j,  0.502624+8.413856e-01j,
E              0.621248-4.769069e-01j,  0.829906-3.271965e-01j],
E            [ 0.403593+3.780714e-01j,  0.280962+3.392000e-02j,...
        A          = array([[[[0.8561369 +0.05969837j, 0.37626302+0.36519146j,
          0.6539293 +0.40403795j, 0.16993557+0.79370755j,
  ...1648j,
          0.50310206+0.6898472j , 0.6573576 +0.9962738j ,
          0.2530085 +0.86227924j]]]], dtype=complex64)
        b          = array([[[[0.04875669+0.31144133j, 0.45005864+0.22624582j,
          0.797893  +0.5535813j , 0.41498896+0.6063427j ],
 ...94375426j, 0.05030875+0.60288817j,
          0.673049  +0.02498303j, 0.7586571 +0.54190135j]]]],
      dtype=complex64)
        bdim       = (2, 3, 5, 4)
        dtype      = <class 'numpy.complex64'>
        lu_and_piv = [array([[[[ 0.78804415+0.9227373j ,  0.69236445+0.7835633j ,
           0.31624767+0.2628439j ,  0.9063425 +0.35367382...,
        [2, 2, 2, 3, 4]],

       [[2, 2, 4, 4, 4],
        [0, 2, 3, 3, 4],
        [2, 1, 4, 3, 4]]], dtype=int32)]
        rng        = Generator(PCG64) at 0x14F7FE85A6C0
        self       = <scipy.linalg.tests.test_batch.TestBatch object at 0x14f888f549b0>
        x          = array([[[[-0.12528789+1.7787557e-03j,  0.5026245 +8.4138554e-01j,
           0.6212484 -4.7690693e-01j,  0.829906  -3....0.14644761+6.8131626e-01j,
          -0.91509163+8.1870073e-01j,  0.05595047-2.2421232e-01j]]]],
      dtype=complex64)
```

### SciPy/NumPy/Python version and system information

```shell
1.16.0.dev0+git20250325.8a3c330 2.2.4 sys.version_info(major=3, minor=13, micro=2, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/local/intel/oneapi/2024/mkl/2024.2/lib/pkgconfig/../../include
    lib directory: /usr/local/intel/oneapi/2024/mkl/2024.2/lib/pkgconfig/../../lib
    name: mkl-dynamic-lp64-iomp
    openblas configuration: unknown
    pc file directory: /usr/local/intel/oneapi/2024/mkl/2024.2/lib/pkgconfig
    version: '2024.2'
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/local/intel/oneapi/2024/mkl/2024.2/lib/pkgconfig/../../include
    lib directory: /usr/local/intel/oneapi/2024/mkl/2024.2/lib/pkgconfig/../../lib
    name: mkl-dynamic-lp64-iomp
    openblas configuration: unknown
    pc file directory: /usr/local/intel/oneapi/2024/mkl/2024.2/lib/pkgconfig
    version: '2024.2'
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.13.6
Compilers:
  c:
    args: -fPIC, -xHost, -fp-model, strict
    commands: icx
    linker: ld.bfd
    linker args: -xHost, -fPIC, -fPIC, -xHost, -fp-model, strict
    name: intel-llvm
    version: 2024.2.0
  c++:
    args: -fp-model=strict, -fPIC, -xHost, -std=c++17
    commands: icpx
    linker: ld.bfd
    linker args: -xHost, -fPIC, -fp-model=strict, -fPIC, -xHost, -std=c++17
    name: intel-llvm
    version: 2024.2.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.12
  fortran:
    args: -xHost, -fPIC, -fp-model, strict
    commands: ifx
    linker: ld.bfd
    linker args: -xHost, -fPIC, -xHost, -fPIC, -fp-model, strict
    name: intel-llvm
    version: 2024.2.0
  pythran: {}
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /discover/nobackup/projects/tm5var4d/packages/python/3.13.2/bin/python3
  version: '3.13'
```

--- a/scipy/linalg/tests/test_batch.py
+++ b/scipy/linalg/tests/test_batch.py
@@ -442,7 +442,7 @@ class TestBatch:
             x = x[..., np.newaxis]
             b = b[..., np.newaxis]
         assert_allclose(A @ x - b, 0, atol=1.5e-6)
-        assert_allclose(x, np.linalg.solve(A, b), atol=2e-6)
+        assert_allclose(x, np.linalg.solve(A, b), atol=3e-6)
 
     @pytest.mark.parametrize('bdim', [(5,), (5, 4), (2, 3, 5, 4)])
     @pytest.mark.parametrize('dtype', floating)
@@ -456,7 +456,7 @@ class TestBatch:
             x = x[..., np.newaxis]
             b = b[..., np.newaxis]
         assert_allclose(A @ x - b, 0, atol=1.5e-6)
-        assert_allclose(x, np.linalg.solve(A, b), atol=2e-6)
+        assert_allclose(x, np.linalg.solve(A, b), atol=3e-6)
 
     @pytest.mark.parametrize('l_and_u', [(1, 1), ([2, 1, 0], [0, 1 , 2])])
     @pytest.mark.parametrize('bdim', [(5,), (5, 4), (2, 3, 5, 4)])

